src/main/kotlin/com/jetbrains/snakecharm/codeInsight/wrapper/WrapperStorage.kt

```diff
@@ -0,0 +1,59 @@
+package com.jetbrains.snakecharm.codeInsight.wrapper
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.util.xmlb.XmlSerializerUtil
+
+@State(name = "WrapperStorage", storages = [Storage("wrapper.xml")])
+class WrapperStorage : PersistentStateComponent<WrapperStorage> {
+
+    var wrapperStringStorage: MutableList<String> = mutableListOf()
+
+    data class Wrapper(
+            val repositoryTag: String,
+            val pathToWrapperDirectory: String,
+            val environmentFileContent: String,
+            val metaFileContent: String,
+            val testSnakefileContent: String
+    ) {
+        fun toList(): List<String> = listOf(
+                repositoryTag,
+                pathToWrapperDirectory,
+                environmentFileContent,
+                metaFileContent,
+                testSnakefileContent
+        )
+    }
+
+    companion object {
+        fun getInstance(): WrapperStorage = ApplicationManager.getApplication().getComponent(WrapperStorage::class.java)
+    }
+
+    fun addWrapper(wrapper: Wrapper) = wrapperStringStorage.addAll(wrapper.toList())
+
+    fun getWrapperList(): List<Wrapper> {
```

Comment:
This method is used by completion contributor, no sense in deserializing the list on each access. Better is to separate serialization/deserialization from actual usage. E.g. make deserialization once and then use a valid object.